tracetypes/org.eclipse.tracecompass.incubator.opentracing.core/src/org/eclipse/tracecompass/incubator/opentracing/core/analysis/callstack/SpanCallStackAnalysis.java

```diff
@@ -0,0 +1,181 @@
+import org.eclipse.tracecompass.incubator.analysis.core.weighted.tree.IDataPalette;
+import org.eclipse.tracecompass.incubator.analysis.core.weighted.tree.IWeightedTreeGroupDescriptor;
+import org.eclipse.tracecompass.incubator.callstack.core.callgraph.CallGraph;
+import org.eclipse.tracecompass.incubator.callstack.core.callgraph.SymbolAspect;
+import org.eclipse.tracecompass.incubator.callstack.core.instrumented.statesystem.CallStackSeries;
+import org.eclipse.tracecompass.incubator.callstack.core.instrumented.statesystem.CallStackStateProvider;
+import org.eclipse.tracecompass.incubator.callstack.core.instrumented.statesystem.InstrumentedCallStackAnalysis;
+import org.eclipse.tracecompass.segmentstore.core.ISegment;
+import org.eclipse.tracecompass.segmentstore.core.ISegmentStore;
+import org.eclipse.tracecompass.statesystem.core.ITmfStateSystem;
+import org.eclipse.tracecompass.tmf.core.exceptions.TmfAnalysisException;
+import org.eclipse.tracecompass.tmf.core.segment.ISegmentAspect;
+import org.eclipse.tracecompass.tmf.core.timestamp.ITmfTimestamp;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ *
+ */
+public abstract class SpanCallStackAnalysis extends InstrumentedCallStackAnalysis {
+
+    private static final String[] DEFAULT_PROCESSES_PATTERN = new String[] { CallStackStateProvider.PROCESSES, "*" }; //$NON-NLS-1$
+
+    private static final List<String[]> PATTERNS = ImmutableList.of(DEFAULT_PROCESSES_PATTERN);
+
+    private @Nullable CallStackSeries fCallStacks;
+
+    private final SpanCallGraphAnalysis fCallGraph;
+
+    private boolean fAutomaticCallgraph=true;
+
+
+    protected SpanCallStackAnalysis() {
+        super();
+        fCallGraph = new SpanCallGraphAnalysis(this);
+    }
+    @Override
+    public boolean setTrace(@NonNull ITmfTrace trace) throws TmfAnalysisException {
+        if (!super.setTrace(trace)) {
+            return false;
+        }
+        return fCallGraph.setTrace(trace);
+    }
+
+    @Override
+    public void setName(String name) {
+        super.setName(name);
+        fCallGraph.setName(name);
+    }
+
+    @Override
+    public synchronized @Nullable CallStackSeries getCallStackSeries() {
+        CallStackSeries callstacks = fCallStacks;
+        if (callstacks == null) {
+            ITmfStateSystem ss = getStateSystem();
+            ITmfTrace trace = getTrace();
+            if (ss == null || trace == null) {
+                return null;
+            }
+            callstacks = new CallStackSeries(ss, getPatterns(), 0, "", getCallStackHostResolver(trace), null); //$NON-NLS-1$
+            fCallStacks = callstacks;
+        }
+        return callstacks;
+    }
+
+
+    /**
+     * Get the patterns for the process, threads and callstack levels in the
+     * state system
+     *
+     * @return The patterns for the different levels in the state system
+     */
+    @Override
+    protected List<String[]> getPatterns() {
```

Comment:
This may be not needed once the patch has been rebased and depends upon the profiling core package.